lib/fluent/plugin_helper/inject.rb

```diff
@@ -132,7 +132,9 @@ def configure(conf)
           if @_inject_time_key
             @_inject_time_formatter = case @inject_config.time_type
                                       when :float then ->(time){ time.to_r.truncate(+6).to_f } # microsecond floating point value
-                                      when :unixtime_millis then ->(time) { time.to_f.floor(3) * 1000 }
+                                      when :unixtime_millis then ->(time) { time.respond_to?(:nsec) ? time.to_i * 1_000 + time.nsec / 1_000_000 : (time * 1_000).floor }
```

Comment:
By this change, this proc will return a different value in some cases as below, but I think the new value is more appropriate:

```
irb(main):001:0> time = Fluent::EventTime.new(1466464211, 320999980)
=> 2016-06-21 08:10:11.320999980 +0900
irb(main):002:0> time.to_f.floor(3) * 1000
=> 1466464211321.0
irb(main):003:0> time.to_i * 1_000 + time.nsec / 1_000_000
=> 1466464211320
```